javatools/build.gradle.kts

```diff
@@ -98,6 +101,23 @@ val jar by tasks.existing(Jar::class) {
     }
 }
 
+/**
+ * TODO: Marcus suggests that merging the jars is not a good way to go. Instead we simply need to
+ *       get the jline.jar into our distribution (installDist task) and point to it in the manifest
+ *       of the javatools.jar
+ */
+val shadowJar by tasks.existing(ShadowJar::class) {
+    dependsOn(jar)
+
+    archiveBaseName = "javatools"
+
+    from(tasks.jar.get().archiveFile)
+
+    dependencies {
+        include(dependency(libs.jline.get()))
+    }
```

Comment:
Could you please do what's "correct" and "right"? I would really appreciate your help

Reply:
I have now done what I think is "correct" and "right". At the moment I remain with javatools.jar as a fat jar including all its dependencies, but I have sketched out a configuration that I can resolve with the application plugin for the xdk distribution in the xdk package, that will make it very easy to detach any dependencies and place them elsewhere, with a Class-Path manifest entry in the javatools.jar artifact. 

The split up solution should be completely invisible to the user, work the same way, and have the advantage that we can start taking stuff out of javatools, instead of watching it grow fatter and fatter. Javatools is already a "god class dependency" given that everything else in the build depends on it. If we compiled javatools.jar and javatools-utils separately, and tried to increase granularity even further, so that a single class change under javatools or javatools utils doesn't necessary require rebuilding the world, we are on the right path.  Fat jars and "god classes" that the entire rest of the world depend on, with no clear separation of its contents, usually mean that 80% of everything gets repackaged, affects the checksums, and throws away any dependency details known, replacing them with the "EVERYTHING" package. 

Homework: please read and understand the lazy map pattern and the distribution and application plugins in the gradle user guide. Given this transform, it should be clear that it's not a very complicated thing, and we have everything to gain from using Gradle built-in plugins, defaults, layouts and expected locations wherever we can.

Reply:
As a side note, specific to jline library, in the future we can trim the "merged in" content. If you look at the [documentation](https://github.com/jline/jline3), there are parts of the library that we definitely have no use for, e.g.: `jline-remote-ssh`, `jline-remote-telnet`, etc. This would require a bit more configuration work...

Reply:
This is typically done by the app vendor distributing the app as different packages, like jline-all, jline-api etc, but in this case they only seem to have a single master artifact on maven central.  Adding modifications to what that artifact contains, as a user, is probably a bad idea. While we do have absolute version control and know that it won't update to a new jline version unless we tell it to, any filter we implement manually will be safe until we change versions. But if we do that, for example because there is a security problem or a feature in the new version that we want, we would have to manually recheck our manual filter, which is potential maintenance cost added.  The artifacts should be considered "unmodifiable" when they are brought in from a repository.

Reply:
> This is typically done by the app vendor distributing the app as different packages, like jline-all, jline-api etc, but in this case they only seem to have a single master artifact on maven central.

This is incorrect; all those libraries do exist separately at the Maven Central:
https://mvnrepository.com/search?q=jline